### ISSUE TITLE
Cylinder head 

### ISSUE THREAD

**Issue by Tergeus** [The-Foundation-Group/airsoft.tools#34]
Hello. When im calculating on your site and others im getting mixed summaries. Its due becouse in your calculator on CYLINDER VOLUME its hard to determine if in your calculation do you need to make corrections on CYLINDER HEAD. Idk if if have that 8 mm (size of cylinder head taking space in cylinder( already in calculation or do i have to make calculations and corrections in field AIRCUSIONS. Idk if that makes sense. It would help if that  cca 8mm cylinder head space was stated if it is in your calculations or not thats all.

**Comment by Arsoth** [comment]
Hey thanks for noting this!

It takes into account a non AOE corrected cyl head for voluming by default, and with the AOE correction button it subtracts an additional 4.5mm of length.

I'll make it clearer that it includes it by default, the airbreak and piston spacers area is mostly for additional volume reduction and because cyl head padding vs piston head padding changes effective port interaction distance (airbreak was admittedly added because I have an airbreak cyl head and wanted to cover that as an option...).

TL;DR is that for the most part so long as you select your cyl type, gear type, and if you're AOE corrected or not it should give you the correct volume.

But I'll make that clearer on the calculator as well!

**Comment by Tergeus** [comment]
OKAY! i got it... 10.5mm in AOE correction was bit too much on first glance (only on piston( + I have EPES Cylinder head that is build with "SLOT" so it takes -1,5mm form depth of cylinder head. IT is space to place sorbopad or PU Rubber. So i have 0 spacers on piston and my AOE is corrected only with cylinder head. AND since PORTED cylinder "doesnt care" about your spacers on piston.... IT was rly hard to make correct calculations. So i had to use AIRSOFTTECH calculator. Your calculator is very simple but this was special case. But thats my bad mostly :D  THANK You bro for your hard work for comunity.

**Comment by Tergeus** [comment]
I have actually one more question when i place AOE corection on PORTED cylinder. PICK UP to CYL ( 10.5mm )should not be taken
into consideration since. The pressure STARTS after piston is past PORT. IT is only should be taken in consideration on CYL 0 and mby 4/5.

**Comment by Arsoth** [comment]
Pickup to back of cyl measures where the mechanically driven part of the piston is sitting in relation to the front of the system. It's not the AOE correction amount (which for "perfect" AOE I measured a bunch of cyl head and piston combos and it is 4.5mm +/- 0.2mm over stock cyl head depth)

There's internal logic to take into account if the AOE is adjusted with the cyl head padding or with piston spacers, which you can test by comparing AOE correction on and off with a ported cyl, and then adding 4.5mm of piston head spacers. Once you tell the system that the AOE correction was on the piston side, it bumps the ratio back up as there's more effective volume (in 99% of the cases).

Ported cyls can matter with piston spacers when short stroking, and especially for DSG/TSG stuff where putting 4.5mm on a piston may shift it between clearing the port and not. For sure an edge case, but something that I wanted to cover.

I did just check my EPeS as well, and it does look like it's about 1mm further back stock as you said. For that one I'd use 7mm on the pickup to cyl distance from my measurement:

![image](https://github.com/user-attachments/assets/0abb7f3c-d38f-4037-9743-f943ff747023)

**Comment by Tergeus** [comment]
Okay i think i got it so you had everything calculated ahead.... well i tried comparing things with 2 calculators. It seems that i have been wrong. A lot of new thing learned for sure. Sorry for bothering you, and thank you for your insights in topic. I am happy that we have guys like you in our "team" <3  i am will correct my adjustments.

**Comment by Arsoth** [comment]
No bother; I'm happy to help people out with any questions! Helps me figure where I need to polish/add documentation as well.

**Comment by Tergeus** [comment]
Funny thing is that i almost never seen your calculator mentioned. Only when topic of springs came out somewhere. I used your calculator with springs. Your callculator is more advanced than other ones. Might be becouse your calculator is showing more accurate and probably lower numbers of compresions. And people might not like it. Ill go with your numbers. I tried AISOFTTECH.Dk calculator i just  did 72mm in lenght - 8.6mm from cylinder head and - 6mm and it showed like 2.70:1 and your calculator showed like 2.60:1 i think or mby even less. It s weird that people are not talking more about your site. It s pity, you did good job there.

**Comment by Arsoth** [comment]
I've been working on it for about a year, but it was only back in July I did a large overhaul of the volume calc to make it better.

Still need to get information on cylinder volumes for springer rifles though...

Almost all of the traffic is driven by word of mouth, so the fact that people are using it at all makes me happy!
